For Android, I am getting this issue https://github.com/mage/mage-sdk-cpp/issues/17, but also:

```
/Users/mt/android-ndk/sources/cxx-stl/gnu-libstdc++/4.6/include/future:111:11: error: declaration of 'struct std::future<Json::Value>'
/Users/mt/Sources/mage-sdk-cpp/platforms/android/jni/../../../src/mage.cpp:49:22: error: 'deferred' is not a member of 'std::launch'
/Users/mt/Sources/mage-sdk-cpp/platforms/android/jni/../../../src/mage.cpp: In member function 'void mage::RPC::SetDomain(std::string)':
/Users/mt/Sources/mage-sdk-cpp/platforms/android/jni/../../../src/mage.cpp:61:3: error: 'httpClient' was not declared in this scope
/Users/mt/Sources/mage-sdk-cpp/platforms/android/jni/../../../src/mage.cpp: In member function 'void mage::RPC::SetApplication(std::string)':
/Users/mt/Sources/mage-sdk-cpp/platforms/android/jni/../../../src/mage.cpp:66:3: error: 'httpClient' was not declared in this scope
/Users/mt/Sources/mage-sdk-cpp/platforms/android/jni/../../../src/mage.cpp: In member function 'void mage::RPC::SetProtocol(std::string)':
/Users/mt/Sources/mage-sdk-cpp/platforms/android/jni/../../../src/mage.cpp:71:3: error: 'httpClient' was not declared in this scope
/Users/mt/Sources/mage-sdk-cpp/platforms/android/jni/../../../src/mage.cpp: In member function 'void mage::RPC::SetSession(std::string)':
/Users/mt/Sources/mage-sdk-cpp/platforms/android/jni/../../../src/mage.cpp:75:3: error: 'httpClient' was not declared in this scope
/Users/mt/Sources/mage-sdk-cpp/platforms/android/jni/../../../src/mage.cpp: In member function 'void mage::RPC::ClearSession()':
/Users/mt/Sources/mage-sdk-cpp/platforms/android/jni/../../../src/mage.cpp:79:3: error: 'httpClient' was not declared in this scope
/Users/mt/android-ndk/sources/cxx-stl/gnu-libstdc++/4.6/include/future: At global scope:
/Users/mt/android-ndk/sources/cxx-stl/gnu-libstdc++/4.6/include/future:143:5: error: 'std::future<typename std::result_of<_Functor(_ArgTypes ...)>::type> std::async(std::launch, _Fn&&, _Args&& ...) [with _Fn = mage::RPC::Call(const string&, const Json::Value&, bool)::<lambda()>, _Args = {}, typename std::result_of<_Functor(_ArgTypes ...)>::type = Json::Value]', declared using local type 'mage::RPC::Call(const string&, const Json::Value&, bool)::<lambda()>', is used but never defined [-fpermissive]
```

--- a/platforms/android/jni/Android.mk
+++ b/platforms/android/jni/Android.mk
@@ -39,6 +39,6 @@ LOCAL_LDLIBS  := -lz -ldl
 LOCAL_WHOLE_STATIC_LIBRARIES += curl_static
 LOCAL_STATIC_LIBRARIES := curl_static_prebuilt
 
-LOCAL_CPPFLAGS := -fexceptions -std=c++11
+LOCAL_CPPFLAGS := -fexceptions -std=c++11 -DHTTP_CONNECTOR
 
 include $(BUILD_STATIC_LIBRARY)

--- a/platforms/android/jni/Application.mk
+++ b/platforms/android/jni/Application.mk
@@ -1,4 +1,5 @@
+NDK_TOOLCHAIN_VERSION := clang
 APP_MODULES := mage-sdk
-APP_STL := gnustl_static
+APP_STL := c++_static
 APP_CPPFLAGS :=  -fexceptions -std=c++11
 APP_PLATFORM := android-12

--- a/platforms/ios/mage-sdk.xcodeproj/project.pbxproj
+++ b/platforms/ios/mage-sdk.xcodeproj/project.pbxproj
@@ -513,6 +513,7 @@
 				DSTROOT = /tmp/mage_sdk.dst;
 				GCC_PRECOMPILE_PREFIX_HEADER = YES;
 				GCC_PREFIX_HEADER = "mage-sdk/mage-sdk-Prefix.pch";
+				GCC_PREPROCESSOR_DEFINITIONS = "HTTP_CONNECTOR=1";
 				HEADER_SEARCH_PATHS = (
 					"$(inherited)",
 					/Applications/Xcode.app/Contents/Developer/Toolchains/OSX10.8.xctoolchain/usr/include,
@@ -524,6 +525,7 @@
 					"$(inherited)",
 					../externals/curl/prebuilt/ios,
 				);
+				OTHER_CPLUSPLUSFLAGS = "$(OTHER_CFLAGS)";
 				OTHER_LDFLAGS = "-ObjC";
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SKIP_INSTALL = YES;
@@ -536,6 +538,7 @@
 				DSTROOT = /tmp/mage_sdk.dst;
 				GCC_PRECOMPILE_PREFIX_HEADER = YES;
 				GCC_PREFIX_HEADER = "mage-sdk/mage-sdk-Prefix.pch";
+				GCC_PREPROCESSOR_DEFINITIONS = "HTTP_CONNECTOR=1";
 				HEADER_SEARCH_PATHS = (
 					"$(inherited)",
 					/Applications/Xcode.app/Contents/Developer/Toolchains/OSX10.8.xctoolchain/usr/include,
@@ -547,6 +550,7 @@
 					"$(inherited)",
 					../externals/curl/prebuilt/ios,
 				);
+				OTHER_CPLUSPLUSFLAGS = "$(OTHER_CFLAGS)";
 				OTHER_LDFLAGS = "-ObjC";
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SKIP_INSTALL = YES;

